### PR TITLE
Update EventCatcherWidget.tid

### PR DESCRIPTION
**Pull request by CodaCodr**
Changed "JavaScript events" to "DOM-initiated Javascript events". 
Removed the quotes from `"event"`. 
Tightened up some of the language.

### CODE REVIEW

File: editions/tw5.com/tiddlers/widgets/EventCatcherWidget.tid [hunk 1]
```diff
@@ -7,16 +7,18 @@ type: text/vnd.tiddlywiki
 ! Introduction
 
 <<.from-version "5.1.23">>
-The event catcher widget traps JavaScript events dispatched within its child content, and allows invoking a series of ActionWidgets in response to the events. 
 
-In order for the events to be trapped they must:
+//This is a technically advanced widget intended for use by those familiar with HTML, CSS and DOM event handling.//
```
Comment (saqimtiaz) [inline editions/tw5.com/tiddlers/widgets/EventCatcherWidget.tid:11]: A few things here. "Technically advanced" is ambigious due to the word technically being used as an adverb. Suggest rewording it to: "This is an advanced technical widget" or dropping the word technical altogether.

As discussed at https://github.com/Jermolene/TiddlyWiki5/discussions/5321#discussioncomment-251219, the use of the term DOM without the context of JavaScript will be confusing to the layman user. We do not explain what DOM is and neither do I think we should. Therefore this needs to be expressed differently.

I suggest something along the lines of "by those familiar with HTML, CSS and JavaScript handling of DOM events".

File: editions/tw5.com/tiddlers/widgets/EventCatcherWidget.tid [hunk 2]
```diff
@@ -7,16 +7,18 @@ type: text/vnd.tiddlywiki
...
+The event catcher widget traps DOM-initiated Javascript events dispatched within its child content, and allows invoking a series of ActionWidgets in response to those events. 
 
-* be of one of the types specified as a parameter to the event catcher widget.
-* arise within a DOM node matching the selector specified as a parameter to the widget.
+In order for the events to be trapped they must:
 
-Use of the event catcher widget is beneficial when using large numbers of other trigger widgets such as the ButtonWidget is causing performance problems. The workflow it enables is akin to what is referred to as "event delegation" in JavaScript parlance. 
+* be of one of the events specified as a parameter to the event catcher widget's `events` attribute.
+* arise within a DOM node specified by the widget's `selector` attribute.
+* support [[event bubbling|https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Building_blocks/Events#Event_bubbling_and_capture]].
```
Comment (saqimtiaz) [inline editions/tw5.com/tiddlers/widgets/EventCatcherWidget.tid:19]: I am unsure what our policy is for linking to external sites for explaining technical terms. This would need input from Jeremy.

File: editions/tw5.com/tiddlers/widgets/EventCatcherWidget.tid [hunk 3]
```diff
@@ -7,16 +7,18 @@ type: text/vnd.tiddlywiki
...
+//This is a technically advanced widget intended for use by those familiar with HTML, CSS and DOM event handling.//
+
+The event catcher widget traps DOM-initiated Javascript events dispatched within its child content, and allows invoking a series of ActionWidgets in response to those events. 
 
-* be of one of the types specified as a parameter to the event catcher widget.
-* arise within a DOM node matching the selector specified as a parameter to the widget.
+In order for the events to be trapped they must:
 
-Use of the event catcher widget is beneficial when using large numbers of other trigger widgets such as the ButtonWidget is causing performance problems. The workflow it enables is akin to what is referred to as "event delegation" in JavaScript parlance. 
+* be of one of the events specified as a parameter to the event catcher widget's `events` attribute.
```
Comment (saqimtiaz) [inline editions/tw5.com/tiddlers/widgets/EventCatcherWidget.tid:17]: Using both the terms parameter and attribute is confusing. Suggest dropping parameter, that is more consistent with other widget documentation.

File: editions/tw5.com/tiddlers/widgets/EventCatcherWidget.tid [hunk 4]
```diff
@@ -7,16 +7,18 @@ type: text/vnd.tiddlywiki
...
+* arise within a DOM node specified by the widget's `selector` attribute.
+* support [[event bubbling|https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Building_blocks/Events#Event_bubbling_and_capture]].
 
-//This is an advanced widget intended to be used by those familiar with HTML, CSS and JavaScript.//
+Use of the event catcher widget is beneficial when using large numbers of other triggering widgets such as the ButtonWidget that may cause performance problems. The event catcher widget alleviates these problems by using an approach very similar to "event delegation" in JavaScript parlance.
```
Comment (saqimtiaz) [inline editions/tw5.com/tiddlers/widgets/EventCatcherWidget.tid:22]: I do not think it is appropriate to state that it alleviates the performance problems. It may help in some situations, but that will vary a lot depending on the architecture and where the actual performance bottleneck is, for instance on render or on refresh. This is why the original documentation is not so specific in this area. This needs to be revised to reflect that.

**Review by saqimtiaz — CHANGES_REQUESTED** [review]
@CodaCodr I've left some comments that need to be addressed before this can be merged.

**Comment by saqimtiaz** [comment]
@CodaCodr to update your PR, find the fork of the Github repo in your account, switch to the same branch as this patch ("patch-1") and just edit the same file again and commit the changes. That will automatically update this patch.

File: editions/tw5.com/tiddlers/widgets/EventCatcherWidget.tid (hunk 3 above)
Comment (CodaCodr) [reply editions/tw5.com/tiddlers/widgets/EventCatcherWidget.tid:17]: Agreed. I'd not have used "parameter" at all had it not been there to begin with.

File: editions/tw5.com/tiddlers/widgets/EventCatcherWidget.tid (hunk 4 above)
Comment (CodaCodr) [reply editions/tw5.com/tiddlers/widgets/EventCatcherWidget.tid:22]: Yes. Agreed. My least liked para of all.

File: editions/tw5.com/tiddlers/widgets/EventCatcherWidget.tid (hunk 3 above)
Comment (saqimtiaz) [reply editions/tw5.com/tiddlers/widgets/EventCatcherWidget.tid:17]: Writing consistent documentation gets tricky when working on multiple things. Filters have parameters and widgets have attributes.

File: editions/tw5.com/tiddlers/widgets/EventCatcherWidget.tid (hunk 2 above)
Comment (CodaCodr) [reply editions/tw5.com/tiddlers/widgets/EventCatcherWidget.tid:19]: To abuse an old adage, I don't think anyone got fired for linking to MDN. But I take your point. I'll remove it.

File: editions/tw5.com/tiddlers/widgets/EventCatcherWidget.tid (hunk 4 above)
Comment (CodaCodr) [reply editions/tw5.com/tiddlers/widgets/EventCatcherWidget.tid:22]: after reviewing again, I see what led me in that direction. The first sentence claims, "that may cause performance problems". This is kind of "out of nowhere" and not supported or countered by the rest of the (original) para. If the remainder does not alleviate/mitigate that claim, it simply serves no purpose, methinks. If the first sentence is removed or otherwise reworded (to not make the claim) then whole thing can be worded either very differently, or dropped. Since I don't know what/where your original intent was headed, I'll reinstate and back off.

**Comment by Jermolene** [comment]
Hi @CodaCodr to my surprise, you haven't yet signed the CLA. With apologies for the bureaucracy, please could you prepare a separate PR to the 'tiddlywiki-com' branch with your signature for the Contributor License Agreement (see [contributing.md](https://github.com/Jermolene/TiddlyWiki5/blob/master/contributing.md)).

**Comment by CodaCodr** [comment]
> Hi @CodaCodr to my surprise,

&lt;gulp> back soon ;)

**Comment by Jermolene** [comment]
Thanks @CodaCodr @saqimtiaz
